stdlib/format.mli

```diff
@@ -1472,6 +1472,79 @@ val kasprintf : (string -> 'a) -> ('b, formatter, unit, 'a) format4 -> 'b
   @since 4.03
 *)
 
+(** Formatted Pretty-Printing with heterogeneous list as arguments.
+
+  The following functions behave the same as their non-'l' counter-parts, but
+  receive a heterogeneous list as arguments.
+
+  These functions provide better error reporting, specifically when the number
+  of arguments doesn't match the format string, and avoid the need for
+  continuation variants, as the list serves as syntactic notion of when the
+  format is finished.
```

Comment:
I would propose to reverse the description and starts with explaining that the heterogeneous list provide a syntactically-delimited notion of arguments for the printers and then describe the advantages of this syntactically-delimited list of argument.


Also I fear that it is not so clear to the reader that "better error reporting" means better error message from the compiler, it is probably better to state explicitly that using those functions often lead to better type error messages in case of a mismatch between the format string and list of arguments.

Reply:
I also agree with the proposed changes here. Let me know if the description that is in place now needs some more clarification